tests/test.py

```diff
@@ -4529,310 +4242,61 @@ class TestZoneIDDecryptionWithVaultMasterKeyLoader(HashiCorpVaultMasterKeyLoader
-        return
+    def get_grpc_channel(self, port):
+        '''setup grpc to use tls client authentication'''
+        with open(TEST_TLS_CA, 'rb') as ca_file, open(TEST_TLS_CLIENT_KEY, 'rb') as key_file, open(TEST_TLS_CLIENT_CERT, 'rb') as cert_file:
+            ca_bytes = ca_file.read()
+            key_bytes = key_file.read()
+            cert_bytes = cert_file.read()
+        tls_credentials = grpc.ssl_channel_credentials(ca_bytes, key_bytes, cert_bytes)
+        return grpc.secure_channel('localhost:{}'.format(port), tls_credentials)
 
-    def setUp(self):
-        self.checkSkip()
+    def get_http_default_kwargs(self):
+        '''setup requests to use client's certificates'''
+        kwargs = super().get_http_default_kwargs()
+        kwargs['verify'] = TEST_TLS_CA
+        # https://requests.readthedocs.io/en/master/user/advanced/#client-side-certificates
+        # first crt, second key
+        kwargs['cert'] = (TEST_TLS_CLIENT_CERT, TEST_TLS_CLIENT_KEY)
+        return kwargs
 
     def apiEncryptionTest(self, request_func, use_http=False, use_grpc=False):
         # one is set
         self.assertTrue(use_http or use_grpc)
         # two is not acceptable
         self.assertFalse(use_http and use_grpc)
         translator_port = 3456
-        connector_port = 12345
-        connector_port2 = connector_port+1
-        client_id = "keypair1"
-        data = get_pregenerated_random_data().encode('ascii')
-        client_id_private_key = read_storage_private_key(KEYS_FOLDER.name, 'keypair1')
-        zone = zones[0]
-        zone_private_key = read_zone_private_key(KEYS_FOLDER.name, zone['id'])
-        connection_string = 'tcp://localhost:{}'.format(translator_port)
-        translator_kwargs = {
-            'incoming_connection_http_string': connection_string if use_http else '',
-            # turn off grpc to avoid check connection to it without acra-connector
-            'incoming_connection_grpc_string': connection_string if use_grpc else '',}
-
-        correct_client_id = 'keypair1'
-        incorrect_client_id = 'keypair2'
-        with ProcessContextManager(self.fork_translator(translator_kwargs)):
-            with ProcessContextManager(self.fork_connector_for_translator(connector_port, translator_port, client_id)):
-                response = request_func(connector_port, correct_client_id, None, data)
-                decrypted = deserialize_and_decrypt_acrastruct(response, client_id_private_key, 'keypair1')
-                self.assertEqual(data, decrypted)
-                # test with correct zone id
-                print("encrypt with zone {}".format(zone['id']))
-                response = request_func(
-                    connector_port, client_id, zone['id'], data)
-                print("decrypt with zone {}".format(zone['id']))
-                decrypted = deserialize_and_decrypt_acrastruct(response, zone_private_key, zone_id=zone['id'].encode('ascii'))
+        key_folder = tempfile.TemporaryDirectory()
```

Comment:
hm, why we don't use same test pattern with existing keys? we have already generated keys for tests from certs. why we re-generate them in temp folder?

Reply:
As I understood, we are generating it in separate temp folder to able to test different `certificate_identifier` like distinguished_name or serial_number in such case this tests looks generic and with help of `TLSAuthenticationBySerialNumberMixin` or `TLSAuthenticationByDistinguishedNameMixin` we can change the logic.

But with the statically declared keys_dir and client_id we will miss such flexibility? What do you think?